code/app/src/main/java/com/example/bread/fragment/HistoryFragment.java

```diff
@@ -1,67 +1,86 @@
-     */
-    // TODO: Rename and change types and number of parameters
-    public static HistoryFragment newInstance(String param1, String param2) {
-        HistoryFragment fragment = new HistoryFragment();
-        Bundle args = new Bundle();
-        args.putString(ARG_PARAM1, param1);
-        args.putString(ARG_PARAM2, param2);
-        fragment.setArguments(args);
-        return fragment;
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+        View view = inflater.inflate(R.layout.fragment_history, container, false);
+
+        moodEventListView = view.findViewById(R.id.historyListView);
+        moodEventArrayList = new ArrayList<>();
+        moodArrayAdapter = new MoodEventArrayAdapter(getContext(), moodEventArrayList);
+        moodEventListView.setAdapter(moodArrayAdapter);
+
+        moodsRepo = new MoodEventRepository();
+        userRepo = new ParticipantRepository();
+
+        fetchParticipantAndLoadEvents();
+
+        return view;
     }
 
-    @Override
-    public void onCreate(Bundle savedInstanceState) {
-        super.onCreate(savedInstanceState);
-        if (getArguments() != null) {
-            mParam1 = getArguments().getString(ARG_PARAM1);
-            mParam2 = getArguments().getString(ARG_PARAM2);
+    //retrieves
+    private void fetchParticipantAndLoadEvents() {
+        FirebaseUser currentUser = FirebaseAuth.getInstance().getCurrentUser(); //https://firebase.google.com/docs/auth/android/manage-users
+        if (currentUser != null) {
+            username = currentUser.getDisplayName();
+            if (username == null) {
+                Log.e("HistoryFragment", "Username is null. Cannot load mood events.");
+                return;
+            }
+            participantRef = userRepo.getParticipantRef(username);
+            loadMoodEvents();
+        } else {
+            Log.e("HistoryFragment", "No authenticated user found.");
         }
     }
 
-    @Override
-    public View onCreateView(LayoutInflater inflater, ViewGroup container,
-                             Bundle savedInstanceState) {
-        // Inflate the layout for this fragment
-        return inflater.inflate(R.layout.fragment_history, container, false);
+    private void loadMoodEvents() {
+        moodsRepo.listenForEventsWithParticipantRef(participantRef, moodEvents -> {
+                    if (moodEvents != null){
+                        moodEventArrayList.clear();
+                        moodEventArrayList.addAll(moodEvents);
+                        //chatGPT prompt "how can i sort an ArrayList of events by timestamp Date object"
+                        moodEventArrayList.sort((e1, e2) -> e2.getTimestamp().compareTo(e1.getTimestamp()));
+                    }
+                    moodArrayAdapter.notifyDataSetChanged();
+                },
+                error -> {
+                    Log.e("MoodHistory", "Failed to listen for mood events", error);
```

Comment:
```suggestion
                    Log.e("HistoryFragment", "Failed to listen for mood events", error);
```
TAG should always stay consistent for a file. its easier to track the error source.
you could add something like `private static final String TAG = "HistoryFragment"` to the class